### PR TITLE
Feat: margin and padding logical props

### PR DESCRIPTION
**Pull request by madhurisandbhor**
### What does it do?

Mapping `margin` and `padding` internally to equivalent logical props.

### Why is it needed?

To support [direction](https://developer.mozilla.org/en-US/docs/Web/CSS/direction) and [writing modes](https://developer.mozilla.org/en-US/docs/Web/CSS/writing-mode) in DS

### How to test it?

You can check logical properties story added for demo purpose(to be improved/updated in RTL feat) where on inspect you can see all margin and padding properties are converted to equivalent logical properties.

### CODE REVIEW

File: packages/strapi-design-system/src/Box/Box.tsx
[the remark this answers is not in the document]
Comment (madhurisandbhor) [reply packages/strapi-design-system/src/Box/Box.tsx:1]: I don't think so, even if you do not use direction/writing mode, its going to be default prop (like paddingTop).

**Review by joshuaellis — CHANGES_REQUESTED** [review]
I think lets take this opportunity to make better typography docs since we should be able to support RTL text.

What do you think of Nasa's typography folder? https://nasa-jpl.github.io/explorer-1/?path=/docs/foundation-typography--fonts Interested to hear if you've found any that you really like :)

**Comment by madhurisandbhor** [comment]
@joshuaellis I did check few design systems for typography folder and I agree we should definitely improve ours. Adobe spectrum also looks good, but Nasa's more simpler like we have with storybooks.
But why don't we do it as a separate PR or with RTL?

**Comment by madhurisandbhor** [comment]
https://canvas.workday.com/tokens/type is interesting too

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**